File: sample-java-app/src/main/java/com/unissey/samplejavaapp/ui/fragments/VideoPlayerFragment.java
package com.unissey.samplejavaapp.ui.fragments;

import static androidx.navigation.Navigation.findNavController;

import android.os.Bundle;
import android.view.View;
import android.widget.MediaController;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import com.unissey.samplejavaapp.R;
import com.unissey.samplejavaapp.databinding.FragmentVideoPlayerBinding;
import com.unissey.samplejavaapp.ui.SharedViewModel;

public class VideoPlayerFragment extends Fragment {

    private FragmentVideoPlayerBinding binding;

    public VideoPlayerFragment() {
        super(R.layout.fragment_video_player);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding = FragmentVideoPlayerBinding.bind(view);
        SharedViewModel sharedViewModel =
                new ViewModelProvider(requireActivity()).get(SharedViewModel.class);

        binding.videoView.setVideoPath(sharedViewModel.getVideoUri().getValue());
        MediaController mediaController = new MediaController(requireContext());
        binding.videoView.setMediaController(mediaController);
        mediaController.setAnchorView(binding.videoView);
        binding.videoView.start();

        binding.restartButton.setOnClickListener(button -> {
            sharedViewModel.setVideoUri("");
            findNavController(requireView()).popBackStack(R.id.demoChoiceFragment, false);
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: sample-legacy-app/src/main/java/com/unissey/samplelegacyapp/ui/fragments/VideoPlayerFragment.kt
package com.unissey.samplelegacyapp.ui.fragments

import android.os.Bundle
import android.view.View
import android.widget.MediaController
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.navigation.fragment.findNavController
import com.unissey.samplelegacyapp.R
import com.unissey.samplelegacyapp.databinding.FragmentVideoPlayerBinding
import com.unissey.samplelegacyapp.ui.SharedViewModel

class VideoPlayerFragment : Fragment(R.layout.fragment_video_player) {

    private var _binding: FragmentVideoPlayerBinding? = null
    private val binding get() = _binding!!

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        _binding = FragmentVideoPlayerBinding.bind(view)

        val sharedViewModel: SharedViewModel by activityViewModels()
        binding.videoView.setVideoPath(sharedViewModel.videoUri.value)

        val mediaController = MediaController(requireContext())
        mediaController.setAnchorView(binding.videoView)
        binding.videoView.setMediaController(mediaController)
        binding.videoView.start()

        binding.restartButton.setOnClickListener {
            sharedViewModel.setVideoUri("")
            findNavController().popBackStack(R.id.demoChoiceFragment, false)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}